File: restdoc-rpc-client/restdoc-dubbo-client/src/main/java/restdoc/client/dubbo/DubboRefBeanManager.java
package restdoc.client.dubbo;

import org.springframework.stereotype.Component;

import java.util.concurrent.ConcurrentHashMap;

@Component
public class DubboRefBeanManager {

    private ConcurrentHashMap<String, Object> refBeanContainer = new ConcurrentHashMap<>();

    public void addRefBean(String beanName, Object bean) {
        refBeanContainer.putIfAbsent(beanName, bean);
    }

    public Object getRefBean(String beanName) {
        return refBeanContainer.get(beanName);
    }
}

File: restdoc-rpc-client/restdoc-dubbo-client/src/main/kotlin/restdoc/client/dubbo/DubboRefBeanManager.kt
package restdoc.client.dubbo

import org.springframework.stereotype.Component
import java.util.concurrent.ConcurrentHashMap

@Component
open class DubboRefBeanManager {

    private val refBeanContainer: ConcurrentHashMap<String, Any> = ConcurrentHashMap()

    open fun addRefBean(beanName: String, bean: Any) {
        refBeanContainer.putIfAbsent(beanName, bean)
    }

    open fun getRefBean(beanName: String): Any {
        return refBeanContainer[beanName]!!
    }
}